### ISSUE TITLE
error[E0277]: the trait bound `tarkov::constant::Item: serde::ser::Serialize` is not satisfied

### ISSUE THREAD

**Issue by rezicG** [dank/tarkov#28]
First off, thank you so much for replying to these issues.

I want to write the items result to a file. I try doing this with serde, however it doesn't seem like it is particularly fond of tarkov::constant::Item .

Is there a way around this?

**Full error message:**
```
error[E0277]: the trait bound `tarkov::constant::Item: serde::ser::Serialize` is not satisfied
    --> src\main.rs:61:36
     |
61   |     serde_json::to_writer(file_items, &items)?;
     |                                       ^^^^^^ the trait `serde::ser::Serialize` is not implemented for `tarkov::constant::Item`
     |
    ::: C:\Users\einarw\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_json-1.0.46\src\ser.rs:2167:8
     |
2167 |     T: Serialize,
     |        --------- required by this bound in `serde_json::ser::to_writer`
     |
     = note: required because of the requirements on the impl of `serde::ser::Serialize` for `std::collections::HashMap<std::string::String, tarkov::constant::Item>`
```

**Cargo version**
`1.41.0`

**Cargo.toml**
```
[package]
name = "test_package"
version = "0.1.0"
authors = ["me"]
edition = "2018"

[dependencies]
serde_json = "1.0"
tarkov = "0.1.1"
tokio = { version = "0.2", features = ["macros"] }
```

**Main.rs**
```
use tarkov::auth::LoginError;
use tarkov::hwid::generate_hwid;
use tarkov::{Error, Tarkov};
use std::io;
//use std::time::{SystemTime, UNIX_EPOCH};
//use tarkov::market_filter::{Currency, MarketFilter, Owner};
use tarkov::profile::Side;
use std::fs::File;
use serde_json;

#[tokio::main]
async fn main() -> Result<(), Error> {
    std::env::set_var("RUST_LOG", "tarkov=info");

    let email = "<email@email>";
    let password = "<pw>;
    let hwid = "<hwid>";
	
    // Create a new hwid, will require 2fa
    //let hwid = generate_hwid();
    //let mut file_hwid = File::create("hwid.txt")?;
    //let byte_hwid = hwid.as_bytes();
    //file_hwid.write_all(byte_hwid)?;

    let t = match Tarkov::login(email, password, &hwid).await {
        Ok(t) => Ok(t),
        Err(Error::LoginError(e)) => match e {
            // 2FA required!
            LoginError::TwoFactorRequired => {
                // Get 2FA from email (or generate TOTP) then continue...
		let mut code = String::new();
		println!("Input 2FA:");
		io::stdin().read_line(&mut code).expect("Failed to read input");
		let len = code.trim_end_matches(&['\r', '\n'][..]).len();
		code.truncate(len);
		println!("{}", &code);
                Tarkov::login_with_2fa(email, password, &code, &hwid).await
            }
            _ => Err(e)?,
        },
        Err(e) => Err(e),
    }?;

    println!("{}", t.session);
	
    let sess = Tarkov::from_session(&t.session);

    // Find and select PMC profile to complete login.
    let profiles = sess.get_profiles().await?;
    let profile = profiles
        .into_iter()
        .find(|p| p.info.side != Side::Savage)
        .unwrap();
    sess.select_profile(&profile.id).await?;

    // Fetch the translation table for market categories.
    let locale = sess.get_i18n("en").await?;



    // Fetch items and write to file	
    let items = sess.get_items().await?;
    let file_items = File::create("items.txt")?;
    serde_json::to_writer(file_items, &items)?;
    Ok(())
}
```

**Comment by rezicG** [comment]
After having fiddled around with the problem for a while, I realized I never needed to serialize the tarkov Item to begin with. I ended up placing the properties I needed into a new hashmap, writing this new hashmap to a file with serde.

I'm pretty new to Rust and keep learning as I go, so I will close this issue. Feel free to comment still, if you have anything to add!

In case anyone else encounters this issue, and wants to know how I went about solving it, here's that part of the code:

```
// Fetch items and write to file	
let items = sess.get_items().await?;
let mut map_items = HashMap::new();
for (k, v) in items {
	map_items.insert(v.name, v.id);
}
let file_items = File::create("items.txt")?;
serde_json::to_writer(file_items, &map_items)?;
```

I get the feeling I was trying to use tarkov::constant::Item in a way it was never meant to be used, so this was probably a non-issue to begin with.
